sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets

```diff
@@ -44,9 +44,25 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
       OverwriteReadOnlyFiles="true" />
 
     <WriteLinesToFile
+        Condition="$(SelfContained)"
         File="$(OutputFile)"
-        Lines="$([System.IO.File]::ReadAllText($(_FunctionsWorkerConfigInputFile)).Replace('$functionExe$', '$(TargetName).exe'))"
+        Lines="$([System.IO.File]::ReadAllText($(_FunctionsWorkerConfigInputFile))
```

Comment:
@brettsam, I think getting rid of `.exe` doesn't work in Windows. I started seeing this error -- `Microsoft.Azure.WebJobs.Script: File DefaultWorkerPath: <redacted>\selfcontained does not exist.`

Wondering if in here we should check RID and based on that, add `.exe` when in Windows.

Reply:
Have the loader do this:
```
if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
{
    fileName += ".exe";
}
```

It's the same thing we do in the functions host to find dotnet. And it looks like we took it from here: https://github.com/dotnet/extensions/blob/04e2b857352b74e3edad5f36abe1ec3abd114a08/shared/Microsoft.Extensions.CommandLineUtils.Sources/Utilities/DotNetMuxer.cs#L42-L45